### ISSUE TITLE
`doc-valid-idents` in `clippy.toml` being ignored

### ISSUE THREAD

**Issue by Kage-Yami** [rust-lang/rust-clippy#8145]
### Summary

I'm attempting to add a new element to `doc-valid-idents` in `clippy.toml` to ignore a GitHub username, but it doesn't seem to be taking any effect, as `clippy:doc_markdown` still flags it.

### Reproducer

I tried this code:

```rust
// ./src/lib.rs
#![warn(clippy::doc_markdown)]

/// TruePikachu
mod example {}
```

```toml
# ./clippy.toml
doc-valid-idents = [
    # defaults
    "KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "DirectX", "ECMAScript", "GPLv2", "GPLv3", "GitHub", "GitLab", "IPv4", "IPv6",
    "ClojureScript", "CoffeeScript", "JavaScript", "PureScript", "TypeScript", "NaN", "NaNs", "OAuth", "GraphQL", "OCaml", "OpenGL",
    "OpenMP", "OpenSSH", "OpenSSL", "OpenStreetMap", "OpenDNS", "WebGL", "TensorFlow", "TrueType", "iOS", "macOS", "FreeBSD", "TeX",
    "LaTeX", "BibTeX", "BibLaTeX", "MinGW", "CamelCase",
    # custom
    "TruePikachu"
]
```

I expected to see this happen:

```
> cargo clippy
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
```

Instead, this happened:

```
> cargo clippy
warning: you should put `TruePikachu` between ticks in the documentation
  --> src\lib.rs:14:5
   |
14 | /// TruePikachu
   |     ^^^^^^^^^^^
   |
note: the lint level is defined here
  --> src\lib.rs:12:9
   |
12 | #![warn(clippy::doc_markdown)]
   |         ^^^^^^^^^^^^^^^^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_markdown

warning: `terraria-minimap-viewer` (lib) generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 0.39s
```

### Version

```text
rustc 1.57.0 (f1edd0429 2021-11-29)
binary: rustc
commit-hash: f1edd0429582dd29cccacaf50fd134b05593bd9c
commit-date: 2021-11-29
host: x86_64-pc-windows-msvc
release: 1.57.0
LLVM version: 13.0.0
```

**Comment by Kage-Yami** [comment]
Oh. It's working now after I nuked `./target`. So perhaps this could be a suggestion to tweak the README to add a comment that cached compilation artifacts can reuslt in this causing false positives if the config is added after the lint originally triggers?

**Comment by xFrednet** [comment]
Thanks for the report, caching currently sadly ignores changes to the configuration. This is the second issue I've seen on the topic. It might be good to document this in the readme until it is fixed :+1:. Would you like to do that?

I'll also add some labels related to fixing the actual underlying issue, of the cache ignoring configuration changes :upside_down_face:

**Comment by Alexendoo** [comment]
Fixed by #9707
